### ISSUE TITLE
Upload shots and thumbnails to new services, and update DB image links

### ISSUE THREAD

**Issue by originalnicodr** [framedsc/HallOfFramed#23]
Upload all of the shots and thumbnails to the service that we chose, grab each image link, and switch the old Discord links for the new ones in the shotsDB. I imagine this will require some bit of Python scripting as well.

I would put the CuratorBot on hold and run the script that reworks the whole DB so as to not have the DB updated while we do this.

Directory-structure-wise, I am thinking of a folder inside the pics folder that has the thumbnails. Another alternative would be putting all the thumbnails in the same folder as the images and adding `thumbnail` to the name of those files.

### Blocked by
- [x] #32 
- [x] #26

**Comment by originalnicodr** [comment]
Since the download [hof script](https://github.com/framedsc/HallOfFramed/issues/30#issuecomment-1837535793) already created the folder and file structure we will be using on Backblaze we decided to use their [CLI tools](https://www.backblaze.com/docs/cloud-storage-command-line-tools) instead of using a python script since it seemed to upload files faster without the risk of the upload being interrupted.

For updating the DB with the image links I wrote the following script:

```python
import os
from tinydb import TinyDB, Query
import urllib.parse as upar
from tqdm import tqdm

def main():
    print("Updating the json DB file with backblaze links.")

    db = TinyDB('shots.json', indent=2)

    for entry in tqdm(list(db)):
        epoch = entry['epochTime']
        shot_filename = upar.urlparse(entry['shotUrl']).path.split('/')[-1]
        entry["shotUrl"] = f"https://cdn.framedsc.com/images/{epoch}_{shot_filename}"

        shot_filename_without_extension = os.path.splitext(shot_filename)[0]
        thumbnail_filename = f"{shot_filename_without_extension}.jpg"

        entry["thumbnailUrl"] = f"https://cdn.framedsc.com/thumbnails/{epoch}_{thumbnail_filename}"

        db.update(entry, Query().epochTime == epoch) 

if __name__ == "__main__":
    main()
```
It is far from optimal since it writes to disk with each update. Could have probably done something better, but it got the job done.
